shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/inline/InlineShardingAlgorithm.java

```diff
@@ -67,7 +68,7 @@ private boolean isAllowRangeQuery() {
     public String doSharding(final Collection<String> availableTargetNames, final PreciseShardingValue<Comparable<?>> shardingValue) {
         Closure<?> closure = createClosure();
         closure.setProperty(shardingValue.getColumnName(), shardingValue.getValue());
-        return closure.call().toString();
+        return getTargetShardingNode(closure, shardingValue.getColumnName());
```

Comment:
@zjcnb Can we add some unit test for this logic?

Reply:
Ok, i think so, How about post good first task?

Reply:
> How about post good fi

@zjcnb Good idea.